lib/checkfunctions.cpp

```diff
@@ -216,13 +216,16 @@ void CheckFunctions::checkIgnoredReturnValue()
             }
 
             if ((!tok->function() || !Token::Match(tok->function()->retDef, "void %name%")) &&
-                (mSettings->library.getUseRetValType(tok) != Library::UseRetValType::NONE ||
+                ((mSettings->library.getUseRetValType(tok) != Library::UseRetValType::NONE) ||
                 (tok->function() && tok->function()->isAttributeNodiscard())) &&
                 !WRONG_DATA(!tok->next()->astOperand1(), tok)) {
                 const Library::UseRetValType retvalTy = mSettings->library.getUseRetValType(tok);
-                if (mSettings->isEnabled(Settings::WARNING) && retvalTy == Library::UseRetValType::DEFAULT)
+                if ((mSettings->isEnabled(Settings::WARNING) &&
```

Comment:
I think this condition is not what you (and I) intend.

warnings should only be written about nodiscard if WARNING is enabled as far as I know.

Reply:
Nice catch. Thanks.